From dd0e38b5feb51c8aa44e76bb6c84202bf8373005 Mon Sep 17 00:00:00 2001
From: Volker Lendecke <vl@samba.org>
Date: Thu, 13 Jun 2013 20:35:32 +0200
Subject: [PATCH] tevent: Fix Coverity ID 989236 Operands don't affect result

"unsigned" could be less than uint64_t, so idx==UINT64_MAX is always false.

Signed-off-by: Volker Lendecke <vl@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>

diff --git a/lib/tevent/tevent_poll.c b/lib/tevent/tevent_poll.c
index c6e2a00ffec6393fd6fccc1a3836f63e4ef4e045..75d0ced5a09ea86505a5df49b3c84073d3aa0f02 100644
--- a/lib/tevent/tevent_poll.c
+++ b/lib/tevent/tevent_poll.c
@@ -546,7 +546,7 @@ static int poll_event_loop_poll(struct tevent_context *ev,
 	   the handler to remove itself when called */
 
 	for (fde = ev->fd_events; fde; fde = fde->next) {
-		unsigned idx = fde->additional_flags;
+		uint64_t idx = fde->additional_flags;
 		struct pollfd *pfd;
 		uint16_t flags = 0;
 
-- 
1.8.2.1

